tensorboard/plugins/scalar/scalars_plugin.py

```diff
@@ -85,11 +117,31 @@ def index_impl(self):
 
   def scalars_impl(self, tag, run, output_format):
     """Result of the form `(body, mime_type)`."""
-    tensor_events = self._multiplexer.Tensors(run, tag)
-    values = [[tensor_event.wall_time,
-               tensor_event.step,
-               tf.make_ndarray(tensor_event.tensor_proto).item()]
-              for tensor_event in tensor_events]
+    if self._db_connection_provider:
+      db = self._db_connection_provider()
+      cursor = db.execute(
+          ('SELECT '
+           'Tensors.computed_time, '
+           'Tensors.data, '
+           'Tensors.step, '
+           'Tensors.dtype, '
+           'FROM Tensors '
+           'LEFT JOIN Tags ON Tensors.series=Tags.tag_id '
+           'LEFT JOIN Runs ON Tags.run_id=Runs.run_id '
+           'WHERE Tensors.step > -1 '
+           'AND Runs.run_name = ? '
+           'AND Tags.tag_name = ? '
+           'AND Tags.plugin_name = ? '),
```

Comment:
Do we want to do an `ORDER BY step` here?  I think we'll need that to handle the reservoir sampled cases where the Tensors will be out-of-order even if they were inserted in step count order, so without it we'd have a regression, although with it we're actually going farther than the existing logdir code since we'd be always sorting by step (which as discussed the logdir code doesn't do if the input steps are out of order, but it probably should).

Reply:
Done. Indeed, this is applicable to recent issues we've seen.